src/transformers/models/pvt_v2/convert_pvt_v2_to_pytorch.py

```diff
@@ -0,0 +1,279 @@
+            state_dict[f"pvt_v2.encoder.layers.{i}.blocks.{j}.attention.value.bias"] = kv_bias[
+                config.hidden_sizes[i] :
+            ]
+
+
+def rename_key(dct, old, new):
+    val = dct.pop(old)
+    dct[new] = val
+
+
+# We will verify our results on an image of cute cats
+def prepare_img():
+    url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+    im = Image.open(requests.get(url, stream=True).raw)
+    return im
+
+
+@torch.no_grad()
+def convert_pvt_v2_checkpoint(pvt_v2_size, pvt_v2_checkpoint, pytorch_dump_folder_path):
+    """
+    Copy/paste/tweak model's weights to our PVT structure.
+    """
+
+    # define default PvtV2 configuration
+    if pvt_v2_size == "b0":
+        config_path = "FoamoftheSea/pvt_v2_b0"
+    elif pvt_v2_size == "b1":
+        config_path = "FoamoftheSea/pvt_v2_b1"
+    elif pvt_v2_size == "b2":
+        config_path = "FoamoftheSea/pvt_v2_b2"
+    elif pvt_v2_size == "b2-linear":
+        config_path = "FoamoftheSea/pvt_v2_b2_linear"
+    elif pvt_v2_size == "b3":
+        config_path = "FoamoftheSea/pvt_v2_b3"
+    elif pvt_v2_size == "b4":
+        config_path = "FoamoftheSea/pvt_v2_b4"
+    elif pvt_v2_size == "b5":
+        config_path = "FoamoftheSea/pvt_v2_b5"
+    else:
+        raise ValueError(
+            f"Available model sizes: 'b0', 'b1', 'b2', 'b2-linear', 'b3', 'b4', 'b5', but "
+            f"'{pvt_v2_size}' was given"
+        )
+    config = PvtV2Config.from_pretrained(config_path)
+    # load original model from https://github.com/whai362/PVT
+    state_dict = torch.load(pvt_v2_checkpoint, map_location="cpu")
+
+    rename_keys = create_rename_keys(config)
+    for src, dest in rename_keys:
+        rename_key(state_dict, src, dest)
+    read_in_k_v(state_dict, config)
+
+    # load HuggingFace model
+    model = PvtV2ForImageClassification(config).eval()
+    model.load_state_dict(state_dict)
+
+    # Check outputs on an image, prepared by PvtImageProcessor
+    image_processor = PvtImageProcessor(size=config.image_size)
+    encoding = image_processor(images=prepare_img(), return_tensors="pt")
+    pixel_values = encoding["pixel_values"]
+    outputs = model(pixel_values)
+    logits = outputs.logits.detach().cpu()
+
+    if pvt_v2_size == "b0":
+        expected_slice_logits = torch.tensor([-1.1939, -1.4547, -0.1076])
+    elif pvt_v2_size == "b1":
+        expected_slice_logits = torch.tensor([-0.4716, -0.7335, -0.4600])
+    elif pvt_v2_size == "b2":
+        expected_slice_logits = torch.tensor([0.0795, -0.3170, 0.2247])
+    elif pvt_v2_size == "b2-linear":
+        expected_slice_logits = torch.tensor([0.0968, 0.3937, -0.4252])
+    elif pvt_v2_size == "b3":
+        expected_slice_logits = torch.tensor([-0.4595, -0.2870, 0.0940])
+    elif pvt_v2_size == "b4":
+        expected_slice_logits = torch.tensor([-0.1769, -0.1747, -0.0143])
+    elif pvt_v2_size == "b5":
+        expected_slice_logits = torch.tensor([-0.2943, -0.1008, 0.6812])
+    else:
```

Comment:
This is specifically for converting the original published ImageNet weights of the PVTv2, and also a hard check to make sure those weights have been copied correctly. Both the PVT and Segformer model in this repository also perform this check, which is where I took the code from.

Reply:
ImageNet weight verification is now opt-in with `--verify-imagenet-weights` option, since I suppose users are far more likely to use this for their own models, since the ImageNet weights are available on the hub.

Commit with fix: https://github.com/huggingface/transformers/pull/26812/commits/d33f0eb492780dd8f1b5519866d1ad2754e81211